pkg/scope/baremetalhost.go

```diff
@@ -128,3 +135,21 @@ func (s *BareMetalHostScope) GetRawBootstrapData(ctx context.Context) ([]byte, e
 
 	return value, nil
 }
+
+// Hostname returns the desired host name.
+func (s *BareMetalHostScope) Hostname() (hostname string) {
+	if s.hasConstantHostname() {
+		hostname = fmt.Sprintf("%s%s-%v", infrav1.BareMetalHostNamePrefix, s.Cluster.Name, s.HetznerBareMetalHost.Spec.ServerID)
+	} else {
+		hostname = infrav1.BareMetalHostNamePrefix + s.HetznerBareMetalHost.Spec.ConsumerRef.Name
+	}
+
+	return hostname
+}
+
+func (s *BareMetalHostScope) hasConstantHostname() bool {
```

Comment:
You could use this, which is a bit shorter:
```
return s.Cluster.GetAnnotations()[infrav1.ConstantBareMetalHostnameAnnotation] == "true"
```
But it does not matter much.

Reply:
I don't think that this method is safe to use in the case where annotations are nil. Probably that will never happen, but that's the only reason for my if statement. Should I remove the check for nil annotations?

Reply:
Getting a value from a nil map just returns the zero value. Setting on a nil map panics.

Yes, I would remove the "if" since it is not needed